project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/settings/AppSettingsConfigurable.kt

```diff
@@ -42,38 +42,23 @@ class AppSettingsConfigurable(private val project: Project) : Configurable {
     }
 
     override fun apply() {
-        project.service<MinimizationPluginSettings>()
-            .stateObservable
-            .apply {
-                var compilationStrategy by compilationStrategy.mutable()
-                compilationStrategy = mySettingsComponent?.compilationStrategy ?: CompilationStrategy.GRADLE_IDEA
-
-                var gradleTask by gradleTask.mutable()
-                gradleTask = mySettingsComponent?.gradleTask ?: "build"
-
-                var gradleOptions by gradleOptions.mutable()
-                gradleOptions = mySettingsComponent?.gradleOptions ?: emptyList()
-
-                var temporaryProjectLocation by temporaryProjectLocation.mutable()
-                temporaryProjectLocation = mySettingsComponent?.temporaryProjectLocation ?: "minimization-project-snapshots"
-
-                var snapshotStrategy by snapshotStrategy.mutable()
-                snapshotStrategy = mySettingsComponent?.snapshotStrategy ?: SnapshotStrategy.PROJECT_CLONING
-
-                var exceptionComparingStrategy by exceptionComparingStrategy.mutable()
-                exceptionComparingStrategy = mySettingsComponent?.exceptionComparingStrategy ?: ExceptionComparingStrategy.SIMPLE
-
-                var stages by stages.mutable()
-                stages = mySettingsComponent?.stages ?: listOf(
-                    FunctionLevelStage(),
-                    FileLevelStage(),
-                )
-
-                var minimizationTransformations by minimizationTransformations.mutable()
-                minimizationTransformations = mySettingsComponent?.transformations ?: listOf(
-                    TransformationDescriptors.PATH_RELATIVIZATION,
-                )
-            }
+        val newState = MinimizationPluginState().apply {
+            compilationStrategy = mySettingsComponent?.compilationStrategy ?: CompilationStrategy.GRADLE_IDEA
+            gradleTask = mySettingsComponent?.gradleTask ?: "build"
+            gradleOptions = mySettingsComponent?.gradleOptions ?: emptyList()
+            temporaryProjectLocation = mySettingsComponent?.temporaryProjectLocation ?: "minimization-project-snapshots"
+            snapshotStrategy = mySettingsComponent?.snapshotStrategy ?: SnapshotStrategy.PROJECT_CLONING
+            exceptionComparingStrategy = mySettingsComponent?.exceptionComparingStrategy ?: ExceptionComparingStrategy.SIMPLE
+            stages = mySettingsComponent?.stages ?: listOf(
+                FunctionLevelStage(),
+                FileLevelStage(),
+            )
+            minimizationTransformations = mySettingsComponent?.transformations ?: listOf(
+                TransformationDescriptors.PATH_RELATIVIZATION,
+            )
```

Comment:
Let's get rid of `?: defaultValue` because default values are defined inside MinimizationPluginState. Instead of it we can just make smth like `mySettingsComponent?.compilationStrategy?.let { compilationStrategy = it }`. Or even better to check mySettingsComponent before updating the state